spec/features/purchase_spec.rb

```diff
@@ -142,15 +142,28 @@
 
         expect(page).to have_field "purchase_line_items_attributes_0_quantity", with: (@existing_barcode.quantity * 2).to_s
       end
-
       scenario "a user can add items that do not yet have a barcode" do
         # enter a new barcode
+        within "#purchase_line_items" do
+          expect(page).to have_xpath("//input[@id='_barcode-lookup-0']")
+          fill_in "_barcode-lookup-0", with: "0\n"
+        end
         # form finds no barcode and responds by prompting user to choose an item and quantity
-        # fill that in
+        expect(page).to have_css("#newBarcode")
+        within "#new_barcode_item" do
+          expect(page).to have_xpath("//input[@id='barcode_item_quantity']")
+          expect(page).to have_xpath("//select[@id='barcode_item_barcodeable_id']")
+          expect(page).to have_field "barcode_item_value", with: "0"
+          # fill that in
+          fill_in "barcode_item_quantity", with: "10"
+          select Item.alphabetized.first.name, from: "barcode_item_barcodeable_id"
```

Comment:
This *probably* won't successfully submit a barcode because no barcode value (`barcode_item_value`) is set -- you can provide any alphanumeric string (no spaces) and it *should* accept it.